api/v1alpha1/managedfusionoffering_types.go

```diff
@@ -0,0 +1,66 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type OfferingKind string
+
+const (
+	KindDataFoundation       OfferingKind = "DF"
+	KindDataFoundationClient OfferingKind = "DFC"
+)
+
+// ManagedFusionOfferingSpec defines the desired state of ManagedFusionOffering
+type ManagedFusionOfferingSpec struct {
+	Kind    OfferingKind      `json:"kind,omitempty"`
```

Comment:
- I think it's better to add `+kubebuilder:validation:Pattern:="(DF|DFC)"` as a validation.

Reply:
After reading about validators I think using Enum is better. Please do let me know if you think otherwise.
Updated in the latest commit.

Reply:
Ack, whatever validation best suites is fine.

Reply:
Ack